src/providers/postgres/qgspgsourceselect.cpp

```diff
@@ -253,6 +253,11 @@ QgsPgSourceSelect::QgsPgSourceSelect( QWidget *parent, Qt::WindowFlags fl, QgsPr
   mProxyModel.setFilterCaseSensitivity( Qt::CaseInsensitive );
   mProxyModel.setSourceModel( &mTableModel );
 
+  // Do not do dynamic sorting - otherwise whenever user selects geometry type / srid / pk columns,
+  // that item suddenly jumps to the end of the list (because the item gets changed) which is very annoying.
+  // The list gets sorted in finishList() method when the listing of tables and views has finished.
+  mProxyModel.setDynamicSortFilter( false );
```

Comment:
Are other providers affected by this too?

Reply:
Just went through the code and it looks like Oracle, MS SQL and DB2 are likely also affected. But I do not have instances of them running here to test and fix, and I would not feel comfortable doing changes blindly.

The issue can be replicated easily: 1. load a connection with a schema containing some tables and at least one view, 2. on the view's row, click on the primary key column, 3. pick a primary key, 4. click away. The view's entry suddenly ends at the end of the list.